test/Core.Test/AdminConsole/Services/PolicyServiceTests.cs

```diff
@@ -341,6 +360,75 @@ await sutProvider.GetDependency<IPolicyRepository>().Received()
         Assert.True(policy.RevisionDate - utcNow < TimeSpan.FromSeconds(1));
     }
 
+    [Theory, BitAutoData]
+    public async Task SaveAsync_ExistingPolicy_UpdateTwoFactor_WithoutMasterPasswordOr2FA_ThrowsBadRequest(
+        [AdminConsoleFixtures.Policy(PolicyType.TwoFactorAuthentication)] Policy policy, SutProvider<PolicyService> sutProvider)
+    {
+        // If the policy that this is updating isn't enabled then do some work now that the current one is enabled
+
+        var org = new Organization
+        {
+            Id = policy.OrganizationId,
+            UsePolicies = true,
+            Name = "TEST",
+        };
+
+        SetupOrg(sutProvider, policy.OrganizationId, org);
+
+        sutProvider.GetDependency<IPolicyRepository>()
+            .GetByIdAsync(policy.Id)
+            .Returns(new Policy
+            {
+                Id = policy.Id,
+                Type = PolicyType.TwoFactorAuthentication,
+                Enabled = false,
+            });
+
+        var orgUserDetailUser = new Core.Models.Data.Organizations.OrganizationUsers.OrganizationUserUserDetails
+        {
+            Id = Guid.NewGuid(),
+            Status = OrganizationUserStatusType.Accepted,
+            Type = OrganizationUserType.User,
+            // Needs to be different from what is passed in as the savingUserId to Sut.SaveAsync
+            Email = "test@bitwarden.com",
+            Name = "TEST",
+            UserId = Guid.NewGuid(),
+            HasMasterPassword = false
+        };
+
+        sutProvider.GetDependency<IOrganizationUserRepository>()
+            .GetManyDetailsByOrganizationAsync(policy.OrganizationId)
+            .Returns(new List<Core.Models.Data.Organizations.OrganizationUsers.OrganizationUserUserDetails>
+            {
+                orgUserDetailUser,
+            });
```

Comment:
I would like this to return multiple orgUsers, with some being eligible for removal (i.e. MP + no 2FA) and some not being eligible (i.e. no MP + no 2FA), in a semi-random order. That way we can make assert that **none** are removed. The idea is to make sure that your `OrderBy` statement in the SUT works as intended.

Reply:
Updated to reflect that!